packages/dom/src/tabbable.js

```diff
@@ -36,18 +36,24 @@ export function isTabbableIndex( element ) {
 	return getTabIndex( element ) !== -1;
 }
 
+/** @typedef {Element & { type?: string, checked?: boolean, name?: string }} MaybeHTMLInputElement */
```

Comment:
As all properties are optional we can use [Partial](https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype) here 

`@typedef {Element &  Partial<{ type: string, checked: boolean, name: string }>} MaybeHTMLInputElement`

Reply:
I like `Partial` for taking an existing named type and casting it to one where all the properties are optional, but when we're defining the type inline I think it's a little cumbersome to read personally.

Reply:
`Partial` makes it straightforward that all props are optional without the need of scanning all the types in an object so for me, it is just easier to read. But it is just a matter of personal opinion so do it as you prefer:)

Reply:
I think a `Partial` here does make things more readable to me, too.

Reply:
This case seems fine as it is. It's a question of preference. I tend to write out these types myself, in my thinking we explicitly write the type we want, rather than writing a different type then transforming it to the desired type. It's mostly stylistic.

Reply:
An alternative to be less verbose here using `Partial` might be something like `type MaybeHTMLInputElement = Partial<HTMLInputElement> & Element`.

Reply:
In https://github.com/WordPress/gutenberg/pull/29220 I extracted a type guard that was helpful there and may be useful here:

https://github.com/WordPress/gutenberg/blob/29601bd2a0427037dde074a376d4b23e2ae61460/packages/dom/src/dom.js#L82-L90

There seems to be a lot of this between the dom file and here.

Reply:
If it's a matter of taste, and doesn't offend anyone too greatly I'll leave this as it is 🙂

Reply:
Totally fine for me to leave as-is FWIW 👍